### ISSUE TITLE
Segmentation fault when parsing pkgdeps with wrong type of comments: '#' i.o. '//'

### ISSUE THREAD

**Issue by jim-bcom** [b-com-software-basis/remaken#30]
**HOW TO REPRODUCE**
* create a packagedependencies.txt file containing only 
```
# This comment does not use the right syntax and causes a crash
```
* run `remaken install` on it

**EXPECTED BEHAVIOR**
* A parsing error of some sort

**OBSERVED BEHAVIOR**
```
Segmentation fault
```

This seems to occur in the `Dependency` ctor, while calling `boost::split` method.

**Comment by stefled** [comment]
Hello

which remaken version is used?

with `remaken 1.10.0 pre-release`, observed behaviour is : 

`[IGNORED]: Dependency line '# This comment does not use the right syntax and causes a crash' with invalid format !`

**Comment by jim-bcom** [comment]
Hi,

I've used remaken 1.10.0, but it was an older release apparently.

I've tested with the latest ( https://github.com/b-com-software-basis/remaken/releases/tag/1.10.0%2Funixes ), and I can reproduce your output.

But does it work as intended ?


* Isn't there supposed to be a syntax for comments, e.g. a line starting with `//` ?
Then, these comments should be ignored, and by ignored, I mean maybe there shouldn't be these `[IGNORED]` message for every comments in the file (ig there was a license header for exemple there'd be multiple useless lines printed for each remaken command invokation.

* In my version, `[IGNORED]` is prompted for a valid comment (line starting with `//`), but other lines seems to be treated as ill-formed dependencies, and crash. Maybe it's due to [this fix](https://github.com/b-com-software-basis/remaken/issues/23) of the latest 1.10.0, but now, it's not that lines starting with `# ` are now treated as comments, it's that some forms of ill-formed deps are ignored, and other forms are not and raise an error. For example with this file:
```
// Comment 1
# wrong comment syntax
/* another wrong comment syntax */
Not commented at all
MyDep|1.0.0|foo
MyDep/1.2.0|MyDep|MyDep@github|https://URL|shared
```
Results in the following output after `remaken install`
```
[IGNORED]: Dependency line '// Comment 1' is commented !
[IGNORED]: Dependency line '# wrong comment syntax' with invalid format !
[IGNORED]: Dependency line '/* another wrong comment syntax */' with invalid format !
[IGNORED]: Dependency line 'Not commented at all' with invalid format !
[IGNORED]: Dependency line 'MyDep|1.0.0|foo' with invalid format !
[2024-02-27 09:25:37.084864] [0x00007fb2ad6ba740] [error]   Dependency file error : invalid repository type : unsupported value= https://URL
Dependency =>
    name=MyDep@github
    version=MyDep
    type=https://URL
    link mode=shared
    identifier=https://URL
    channel=stable

[2024-02-27 09:25:37.084973] [0x00007fb2ad6ba740] [error]   Error parsing dependency file : invalid format
```

To summarize:
* proper comments (`//` currently I think, although I find `#` more natural for this kind of file, maybe being used to Bash) should be silently ignored, as any comment is in other tools. Or at the very least, only prompted in verbose mode, but even then, I'm not sure.
* Any other uncommented lines should be considered as a dependency description, and thus a error should be raised when ill-formed (as it is the case for the last example.

What do you think ?

Also, I've tried to add a comment at the end of a well-formed dep, but it seems to be interpreted as a `mode`:
```
[WARNING]: Current mode 'shared // plop' is unsupported for 'MyDeps' dependency, then apply 'shared'default mode
```

**Comment by jim-bcom** [comment]
Sorry, I just noticed that comments and ill-formed deps are indeed treated diffferently in the last version
```
[IGNORED]: Dependency line '// Comment 1' is commented !
[IGNORED]: Dependency line '# wrong comment syntax' with invalid format !
```

So, for these two case, to me, the first line shouldn't be printed (or at least in verbose mode), and second line et alike maybe shouldn't have `[IGNORED]` printed.

What do you think ?

**Comment by stefled** [comment]
The current way to comment a dependency is to start with `//` and this is actually displayed to the developer.
It is possible to put it only in verbose mode, but that seems dangerous/useless to me. 

For me comments or an ignored lign in a `packagedependencies.txt` file should only be used for local testing purposes, and I find it useful to systematically trace whether a dependency is in comments.

In the case of a ill-formatted dependency which starts with # or other, I chose to ignore the dependency and not to stop the execution with an error.
I can change to raise an error, but the ignore message text is explicit

**Comment by stefled** [comment]
in order to display less messages to users, the [IGNORED] dependencies comments are only displayed in verbose mode

wil be available in next 1.11.0 version
